govcd/system.go

```diff
@@ -975,3 +975,41 @@ func (vcdCli *VCDClient) GetOrgList() (*types.OrgList, error) {
 	}
 	return orgList, nil
 }
+
+// QueryAdminOrgVdcStorageProfileByID finds a StorageProfile of VDC by ID as admin
+func QueryAdminOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) (*types.QueryResultAdminOrgVdcStorageProfileRecordType, error) {
+	results, err := vcdCli.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":          types.QtAdminOrgVdcStorageProfile,
+		"filter":        fmt.Sprintf("id==%s", url.QueryEscape(id)),
+		"filterEncoded": "true",
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(results.Results.AdminOrgVdcStorageProfileRecord) == 0 {
+		return nil, ErrorEntityNotFound
+	}
+	if len(results.Results.AdminOrgVdcStorageProfileRecord) > 1 {
+		return nil, fmt.Errorf("more than one Storage Profile found with ID %s", id)
+	}
+	return results.Results.AdminOrgVdcStorageProfileRecord[0], nil
+}
+
+// QueryOrgVdcStorageProfileByID finds a StorageProfile of VDC by ID
+func QueryOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) (*types.QueryResultOrgVdcStorageProfileRecordType, error) {
```

Comment:
Missing test for this function.

Moreover, since we are reviewing this query type, the original structure – which was added in 2016 by another team – is missing several fields. 
Here is what I get when fetching an `OrgVdcStorageProfileRecord`
```xml
<OrgVdcStorageProfileRecord
diskIopsDefault="0"
diskIopsEnabled="false"
diskIopsMax="0"
diskIopsPerGbMax="0"
iopsAllocated="0"
iopsLimit="0"
isDefaultStorageProfile="true"
isEnabled="true"
isVdcBusy="false"
name="*"
numberOfConditions="0"
storageLimitMB="0"
storageUsedMB="34817"
vdc="https://example.com/api/vdc/677720eb-28b0-4bab-b108-38f4a2238db2"
vdcName="vdc-datacloud"
href="https://example.com/api/vdcStorageProfile/fb2c5d4f-78f4-4b49-afa5-6b34c79de529"/>
```
Several fields are missing from the original declaration. When you add the test for this function, you should either add such fields, or provide a good reason for skipping them.

Reply:
Add tests and fields, but look please my last comment